apps/test-viewer/src/components/ViewerOptions.tsx

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { createContext, PropsWithChildren, useContext, useState } from "react";
+import { StatusBarItem, StatusBarSection, UiItemsProvider } from "@itwin/appui-react";
+import { SvgVisibilityShow } from "@itwin/itwinui-icons-react";
+
+export interface ViewerOptionsContext {
+  density: "default" | "enlarged";
+}
+
+export interface ViewerActionsContext {
+  setDensity: React.Dispatch<React.SetStateAction<"default" | "enlarged">>;
+}
+
+const viewerOptionsContext = createContext<ViewerOptionsContext>({} as ViewerOptionsContext);
+const viewerActionsContext = createContext<ViewerActionsContext>({} as ViewerActionsContext);
+
+export function ViewerOptionsProvider(props: PropsWithChildren<unknown>) {
+  const [density, setDensity] = useState<"default" | "enlarged">("default");
+
+  return (
+    <viewerActionsContext.Provider value={{ setDensity }}>
+      <viewerOptionsContext.Provider value={{ density }}>{props.children}</viewerOptionsContext.Provider>
+    </viewerActionsContext.Provider>
+  );
+}
+
+export function useViewerOptionsContext() {
+  return useContext(viewerOptionsContext);
+}
+
+export function useViewerActionsContext() {
+  return useContext(viewerActionsContext);
+}
+
+export class ViewerOptionsUiItemsProvider implements UiItemsProvider {
```

Comment:
Played around with test-viewer and it was refreshing whole app on each layout toggle. Creating custom button component allows to avoid refreshing whole app:
```tsx
export const statusBarActionsProvider: UiItemsProvider = {
  id: "ViewerOptionsUiItemsProvider",
  getStatusBarItems: () => [
    {
      id: `expandedLayoutButton`,
      tooltip: "Toggle expanded layout",
      content: <ToggleLayoutButton />,
      itemPriority: 1,
      execute: () => {},
      section: StatusBarSection.Left,
    },
  ],
};

function ToggleLayoutButton() {
  const { setDensity } = useViewerActionsContext();
  return (
    <IconButton styleType="borderless" onClick={() => setDensity((prev) => (prev === "default" ? "enlarged" : "default"))}>
      <SvgVisibilityShow />
    </IconButton>
  );
}
```

Reply:
Applied the suggestion, however it was rendering two buttons so made some additional changes.